source/Ship.cpp

```diff
@@ -564,9 +549,12 @@ void Ship::FinishLoading(bool isNewInstance)
 	// Check that all the "equipped" weapons actually match what your ship
 	// has, and that they are truly weapons. Remove any excess weapons and
 	// warn if any non-weapon outfits are "installed" in a hardpoint.
+	auto equipped = armament.GetEquipped();
```

Comment:
You'd then be doing something like this here:
```c++
const auto equipped = std::accumulate(Weapons().begin(), Weapons().end(), [](map<const Outfit *, unsigned> m, const Hardpoint &h)
{
  if((auto *o = h.GetOutfit()))
    ++m[o];
  return m;
}, map<const Outfit *, unsigned>{});
```
I think making the map argument to the lambda take rvalues and doing `return std::move(m);` is a "pessimizing move" antipattern, but it'd be worth verifying that. (It's not required to use accumulate here.)

Reply:
I added a GetEquipped() helper function to the anonymous namespace of the Ships-class, since the functionality is called twice from different functions within Ship.
For readability I used a for-loop; your code is very cool, but I feel that the readability/maintainability of a regular for-loop is better than an accumulate that uses lambda functions.